src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/ConfigurationSetting.cs

```diff
@@ -93,12 +93,18 @@ public ConfigurationSetting(string key, string value, string label = null)
         public bool Equals(ConfigurationSetting other)
         {
             if (other == null) return false;
+            if (ETag != null && other.ETag != null)
```

Comment:
What if we just drop this `if` guard?

Reply:
I think two settings should compare equal if one was created on the client and one came from the server as long as all their "logical" fileds are equal. But then the one created on the client won't have the etag.

Reply:
I guess "logical" means all fields that are not `readOnly` or `writeOnly` in [swagger lingo](https://swagger.io/docs/specification/data-models/data-types/). Which sounds reasonable. Especially if the etag is not based on the value of the resource (but also includes things like sequence numbers etc.)

Reply:
If I understand correctly what you guys are saying, this approach covers the concerns. Is that correct?

Reply:
Ping @KrzysztofCwalina @johanste